shell/platform/fuchsia/flutter/session_connection.cc

```diff
@@ -129,6 +131,40 @@ void SessionConnection::OnSessionSizeChangeHint(float width_change_factor,
                                              height_change_factor);
 }
 
+fml::TimePoint SessionConnection::CalculateNextLatchPoint(
+    fml::TimePoint present_requested_time,
+    fml::TimePoint now,
+    fml::TimePoint last_latch_point_targeted,
+    fml::TimeDelta flutter_frame_build_time,
+    fml::TimeDelta vsync_interval,
+    std::deque<std::pair<fml::TimePoint, fml::TimePoint>>&
+        future_presentation_infos) {
+  // The minimum latch point is the largest of:
+  // Now
+  // When we expect the Flutter work for the frame to be completed
+  // The last latch point targeted
+  fml::TimePoint minimum_latch_point_to_target =
+      std::max(std::max(now, present_requested_time + flutter_frame_build_time),
+               last_latch_point_targeted);
+
+  auto it = future_presentation_infos.begin();
+  while (it != future_presentation_infos.end()) {
+    fml::TimePoint latch_point = it->first;
+
+    if (latch_point >= minimum_latch_point_to_target) {
+      return latch_point;
+    }
+
+    it++;
+  }
+
+  FML_DCHECK(it == future_presentation_infos.end());
```

Comment:
nit: I don't think that last DCCHECK is very useful. Maybe use this instead:

for (auto& info : future_presentation_infos) {
  ...
}

Reply:
Done